File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        vector<vector<string>> ans;
        unordered_map<string,int> mp;
        for(int i = 0 ; i < strs.size() ; i++){
            string s(strs[i]);
            sort(s.begin(),s.end());
            if(mp.find(s) == mp.end()){
                mp[s] = ans.size();
                ans.push_back({});
            }
            ans[mp[s]].push_back(strs[i]);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> ans = new ArrayList<>();
        Map<String,Integer> mp = new HashMap<>();
        for(int i = 0 ; i < strs.length ; i++){
            char[] charArray = strs[i].toCharArray();
            Arrays.sort(charArray);
            String s = Arrays.toString(charArray);
            if(mp.get(s) == null){
                mp.put(s,ans.size());
                ans.add(new ArrayList<>());
            }
            ans.get(mp.get(s)).add(strs[i]);
        }
        return ans;
    }
}